pkgs/build-support/node/build-npm-package/hooks/npm-install-hook.sh

```diff
@@ -34,7 +34,16 @@ npmInstallHook() {
 
     if [ ! -d "$nodeModulesPath" ]; then
         if [ -z "${dontNpmPrune-}" ]; then
-            npm prune --omit=dev --no-save ${npmWorkspace+--workspace=$npmWorkspace} $npmPruneFlags "${npmPruneFlagsArray[@]}" $npmFlags "${npmFlagsArray[@]}"
+            if ! npm prune --omit=dev --no-save ${npmWorkspace+--workspace=$npmWorkspace} $npmPruneFlags "${npmPruneFlagsArray[@]}" $npmFlags "${npmFlagsArray[@]}"; then
+              echo
+              echo
+              echo 'ERROR: `npm prune --omit=dev` failed'
+              echo
+              echo 'If npm tried to download additional dependencies above, try setting `dontNpmPrune = true`'
+              echo
```

Comment:
```suggestion
              echo
              echo
              echo "ERROR: npm prune step failed"
              echo
              echo 'If npm tried to download additional dependencies above, try setting `dontNpmPrune = true`.'
              echo
```

Maybe don't quote a command that may be wrong if someone sets, e.g. `npmPruneFlags = ["--something"]`, and gets confused why the error message doesn't match what their command is supposed to be

Also could you update the commit message to `buildNpmPackage: ...` so that it's not missed when something searches the commit log for `buildNpmPackage`?

Otherwise this looks lovely, thank you!

Reply:
Makes total sense, I should have come up with that myself. Thanks. Ammended.